sdk/eidolon_ai_sdk/agent/simple_agent.py

```diff
@@ -26,7 +27,11 @@ class ActionDefinition(BaseModel):
     user_prompt: str = "{{ body }}"
     input_schema: dict = None
     output_schema: Union[Literal["str"], Dict[str, Any]] = "str"
-    files: Literal["disable", "single-optional", "single", "multiple"] = "disable"
+    allow_file_upload: bool = False
+    # allow all types for text, image, audio, word, pdf, json, etc
+    supported_mime_types: Literal[""] = ["application/json", "text/plain", "image/*", "audio/*", "application/pdf", "application/msword", "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
```

Comment:
This won't work. The default is not checked against the type, which is why it is currently working.

You want List[Literal[...]] In the typing.

Or perhaps even List[str] with a warning about an unsupported mime type